deployment/configMigration.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+
+from clusterObjectModel import forward_compatibility
+import shutil
+import yaml
+import os
+import sys
+print("This script is used for migrating config from v0.8 to v0.10!")
+print("Usage: configMigration.py from_directory to_directory")
+
+input_dir = os.path.expanduser(sys.argv[1])
+output_dir = os.path.expanduser(sys.argv[2])
+print "Input directory:", input_dir
+print "Output directory:", output_dir
+
+# generate layout.yaml
+print("Migrating layout.yaml...")
+old_cluster_config_file = os.path.join(input_dir, "cluster-configuration.yaml")
+old_cluster_config = yaml.load(open(old_cluster_config_file))
+default_properties = old_cluster_config["default-machine-properties"]
+layout = {"machine-list": []}
+for node in old_cluster_config["machine-list"]:
+    new_node_config = {}
+    new_node_config.update(default_properties)
+    new_node_config.update(node)
+    new_node_config["nodename"] = node["hostip"]
+    layout["machine-list"].append(new_node_config)
+layout["machine-sku"] = old_cluster_config["machine-sku"]
+# read api-server ip from kubernetes-configuration.yaml
+kubernetes_config = yaml.load(open(os.path.join(input_dir, "kubernetes-configuration.yaml")))
+# got dashboard ip
+dashboard_ip = ""
+for node in old_cluster_config["machine-list"]:
+    if ("dashboard" in node and (node["dashboard"] == "true")):
+        dashboard_ip = node["hostip"]
+        break
+# assign api-servers-url and dashboard-url
+layout["kubernetes"] = {
+    "api-servers-url": "http://"+kubernetes_config["kubernetes"]["load-balance-ip"]+":443",
+    "dashboard-url": "http://"+dashboard_ip+":80"}
+
+# write out the file
+layout_file = os.path.join(output_dir, "layout.yaml")
+if(os.path.exists(layout_file)):
+    print"!!! Error, file already existing:", layout_file
+    exit(-1)
```

Comment:
sys.exit(-1).
[https://stackoverflow.com/questions/6501121/difference-between-exit-and-sys-exit-in-python](https://stackoverflow.com/questions/6501121/difference-between-exit-and-sys-exit-in-python)

Reply:
Great.

Reply:
Hi @ydye ,

One more question, one more disadvantage is that:

> If I use exit() in a code and run it in the shell, it shows a message asking whether I want to kill the program or not. It's really disturbing.

But I haven't see it. Maybe I have been using the sys.exit()? Because I had imported sys.
Any suggestion?

Reply:
I have no idea about this, maybe only happened in a desktop?